data-model/src/main/java/io/micronaut/data/model/JsonType.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.data.model;
+
+/**
+ * Enum of JSON type representations or actually storage types. It is useful to know how to store and retrieve JSON fields.
+ *
+ * @author radovanradic
+ * @since 4.0.0
+ * @see PersistentProperty#getJsonType()
+ */
+public enum JsonType {
```

Comment:
I think `JsonDataType` would be a better naming

Reply:
Changed